src/python/pants/invalidation/build_invalidator.py

```diff
@@ -58,7 +58,9 @@ def __init__(self, cache_key_gen_version=None):
     cache_key_gen_version - If provided, added to all cache keys. Allows you to invalidate
       all cache keys in a single pants repo, by changing this value in config.
     """
-
+    # TODO: Qualify the BuildInvalidator root with the cache_key_gen_version, instead of mixing
+    # it into the key, for uniformity with other qualifiers like the fprint strategy and the
```

Comment:
Mixing this into the key is necessary for use in the buildcache though I think. Unlike the strategy (which should implicitly result in different keys due to logical differences), the cache_key_gen_version has no purpose _other_ than to be mixed in the key.